client/db.go

```diff
@@ -551,17 +563,47 @@ func (b *Batch) fillResults() error {
 			case *proto.DeleteResponse:
 				row := &result.Rows[k]
 				row.Key = []byte(call.Args.(*proto.DeleteRequest).Key)
-			case *proto.DeleteRangeResponse:
-			case *proto.EndTransactionResponse:
+
 			case *proto.AdminMergeResponse:
 			case *proto.AdminSplitResponse:
+			case *proto.DeleteRangeResponse:
+			case *proto.EndTransactionResponse:
+			case *proto.InternalBatchResponse:
+			case *proto.InternalGCResponse:
+			case *proto.InternalMergeResponse:
+			case *proto.InternalPushTxnResponse:
+			case *proto.InternalRangeLookupResponse:
+			case *proto.InternalResolveIntentResponse:
+				// Nothing to do for these methods as they do not generate any
+				// rows. For the proto.Internal* responses the caller will have hold of
+				// the response object itself.
+
 			default:
-				return fmt.Errorf("unsupported reply: %T", call.Reply)
+				if result.Err == nil {
+					result.Err = fmt.Errorf("unsupported reply: %T", call.Reply)
+				}
 			}
 		}
 		offset += result.calls
 	}
-	return nil
+}
+
+// InternalAddCall adds the specified call to the batch. It is intended for
+// internal use only. It is an error to use InternalAddCall to execute
+// operations that are available via the Batch methods (e.g. Get).
+func (b *Batch) InternalAddCall(call Call) {
```

Comment:
Yep, what @tschottdorf says is correct. I'd like to prevent anyone else from calling this, but go doesn't offer a means of doing so right now. I think `internal` packages might help eventually, but they aren't available yet.